lib/utils/helpers.ts

```diff
@@ -187,3 +187,27 @@ export function getSubShellCommand(command: string) {
 		};
 	}
 }
+
+export function retry<T>(
+	func: () => T,
+	times: number,
+	label: string,
+	delayMs = 1000,
+	factor = 2,
+	count = 0,
+): Bluebird<T> {
+	let promise = Bluebird.try(func);
+	if (count < times) {
+		promise = promise.catch((err: Error) => {
+			const delay = factor ** count * delayMs;
+			console.log(
+				`Retrying "${label}" after ${(delay / 1000).toFixed(2)}s (${count +
+					1} of ${times}) due to: ${err}`,
+			);
+			return Bluebird.delay(delay).then(() =>
+				retry(func, times, label, delayMs, factor, count + 1),
```

Comment:
I'm missing something here, why doesn't it `return` the retry promise?

Reply:
Do you mean `return retry()` as opposed to `retry()`? Note there are no curly brackets, so the `return` statement is not needed. Compare:
* `() => retry()`
* `() => { return retry(); }`

Reply:
Ah yes, I'd missed the lack of curly braces. My fault. :)